agent/config/config.go

```diff
@@ -373,20 +400,37 @@ func ec2MetadataConfig(ec2client ec2.EC2MetadataClient) Config {
 // error is returned, however, if the config is incomplete in some way that is
 // considered fatal.
 func NewConfig(ec2client ec2.EC2MetadataClient) (config *Config, err error) {
-	ctmp := environmentConfig() //Environment overrides all else
+	var errs []error
+	ctmp, env_err := environmentConfig() //Environment overrides all else
+	if env_err != nil {
```

Comment:
I think we'd better follow the naming conventions in golang(without underscore)? multiple places